app/util/PlutoMessageConsumer.scala

```diff
@@ -0,0 +1,70 @@
+package util
+
+import data.UnpackedDataStores
+import akka.actor.{ActorSystem, Scheduler}
+import scala.annotation.tailrec
+import scala.concurrent.duration._
+import play.api.libs.json._
+import com.amazonaws.services.sqs.model.{DeleteMessageRequest, ReceiveMessageRequest, Message}
+import scala.concurrent.ExecutionContext
+import com.gu.media.logging.Logging
+import data.DataStores
+import scala.collection.JavaConverters._
+
+case class PlutoMessageConsumer(val stores: DataStores, awsConfig: AWSConfig)
+  extends UnpackedDataStores with Logging {
+
+  val actorSystem = ActorSystem("PlutoMessageConsumer")
+
+
+  def start(scheduler: Scheduler)(implicit ec: ExecutionContext): Unit = {
+    log.info("Starting uploads sqs queue reader")
+    actorSystem.scheduler.scheduleOnce(0.seconds)(processMessages())
+
+  }
+
+  @tailrec
+  private def processMessages(): Unit = {
+
+    for (
+      msg <- getMessages(waitTime = 20, maxMessages = 1)
+    ) yield {
+      handleMessage(msg)
+      awsConfig.sqsClient.deleteMessage(new DeleteMessageRequest(awsConfig.plutoQueueUrl, msg.getReceiptHandle()))
+    }
+
+    processMessages()
+
+  }
+
+  private def getMessages(waitTime: Int, maxMessages: Int): Seq[Message] =
+    awsConfig.sqsClient.receiveMessage(
+      new ReceiveMessageRequest(awsConfig.plutoQueueUrl)
+        .withWaitTimeSeconds(waitTime)
+        .withMaxNumberOfMessages(maxMessages)
+    ).getMessages.asScala.toList
+
+  private def handleMessage(msg: Message): Unit = {
+
+    val body = Json.parse(msg.getBody)
```

Comment:
I'm not sure this allows for extension of `PlutoMessage`, since you're passing in the string.

Maybe something like:

```scala
val json = Json.parse(msg.getBody)

(json \ "Messsage").validate[PlutoMessage] match {
  case JsSuccess(message, _) =>
     // do stuff

  case JsError(errors) =>
    log.error("Could not extract a message body blah blah")
```